### PR TITLE
Update readme for DPS support

### PR DESCRIPTION
**Pull request by yzhong94**
Adding feature matrix for DPS service + device SDK

### CODE REVIEW

**Comment by yzhong94** [comment]
Please review the device matrix carefully.  Did testing for MQTT-ws and AMQP-ws (X.509) complete?

File: readme.md [hunk 1]
```diff
@@ -60,6 +60,28 @@ Devices and data sources in an IoT solution can range from a simple network-conn
...
+| TPM Individual Enrollment   | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark:* | :heavy_check_mark: | This SDK supports connecting your device to the Device Provisioning Service via [individual enrollment](https://docs.microsoft.com/en-us/azure/iot-dps/concepts-service#enrollment) using [Trusted Platform Module](https://docs.microsoft.com/en-us/azure/iot-dps/concepts-security#trusted-platform-module-tpm).  This [quickstart](https://docs.microsoft.com/en-us/azure/iot-dps/quick-create-simulated-device) reviews how to create a simulated device for individual enrollment with TPM. AMQP connection over Websocket is in progress.                                                                                                                                                                                                               |
```
Comment (tameraw) [inline readme.md]: >:heavy_check_mark: [](start = 74, length = 18)

This is also not available yet for the same reason as WS.

**Comment by yzhong94** [comment]
Fixed the error Tamer pointed out.  Please merge this!

File: readme.md [hunk 2]
```diff
@@ -60,6 +60,29 @@ Devices and data sources in an IoT solution can range from a simple network-conn
...
+| TPM Individual Enrollment   | :heavy_check_mark: | :heavy_multiplication_x: | :heavy_check_mark: | :heavy_multiplication_x: | :heavy_check_mark: | This SDK supports connecting your device to the Device Provisioning Service via [individual enrollment](https://docs.microsoft.com/en-us/azure/iot-dps/concepts-service#enrollment) using [Trusted Platform Module](https://docs.microsoft.com/en-us/azure/iot-dps/concepts-security#trusted-platform-module-tpm).  This [quickstart](https://docs.microsoft.com/en-us/azure/iot-dps/quick-create-simulated-device) reviews how to create a simulated device for individual enrollment with TPM. Websocket connection is currently not available.                                                                                                                                                                                                               |
```
Comment (prmathur-microsoft) [inline readme.md]: Actually TPM over MQTT is not supported. Can you change that ?

Comment (prmathur-microsoft) [reply readme.md]: This is true for all SDK's. MQTT only has X509 support.

Comment (prmathur-microsoft) [reply readme.md]: You also specififcally say AMQP_WS is work in progress.

File: readme.md [hunk 3]
```diff
@@ -60,6 +60,29 @@ Devices and data sources in an IoT solution can range from a simple network-conn
...
+| X.509 Enrollment Group      | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | This SDK supports connecting your device to the Device Provisioning Service via [enrollment group](https://docs.microsoft.com/en-us/azure/iot-dps/concepts-service#enrollment) using [X.509 root certificate](This SDK supports connecting your device to the Device Provisioning Service via [individual enrollment](https://docs.microsoft.com/en-us/azure/iot-dps/concepts-service#enrollment) using [X.509 leaf certificate](https://docs.microsoft.com/en-us/azure/iot-dps/concepts-security#leaf-certificate))                                                                                                                                                                                            |
```
Comment (prmathur-microsoft) [inline readme.md]: Can you also point to samples folder here ? like you did with service SDK ?

File: readme.md (hunk 2 above)
Comment (yzhong94) [reply readme.md]: TPM over MQTT is not supported on all SDKs?  @tameraw can you confirm that?

**Comment by prmathur-microsoft** [comment]
Once you update TPM support over MQTT and squash all the commits let me know and I will merge it in.

**Comment by yzhong94** [comment]
@prmathur-microsoft Please take a look!  If this looks good, I will merge the commits.

**Comment by prmathur-microsoft** [comment]
Thanks @yzhong94 for taking care of this. I have approved changes.

**Pull request merged**